lib/slavery.rb

```diff
@@ -18,8 +18,16 @@ def spec_key
       @spec_key ||= "#{ActiveRecord::ConnectionHandling::RAILS_ENV.call}_slave"
     end
 
-    def on_slave(&block)
-      Base.new(:slave).run &block
+    def slave_connections
+      @slave_connections ||= {}
+    end
+
+    def on_slave(name = nil, &block)
```

Comment:
This method shouldn't have any `if` statements. It should be handled inside `Base`.

Reply:
This should be addressed now.